parsers/esdtTransferParser.go

```diff
@@ -0,0 +1,181 @@
+package parsers
+
+import (
+	"bytes"
+	"errors"
+	"math/big"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/ElrondNetwork/elrond-vm-common/check"
+	"github.com/ElrondNetwork/elrond-vm-common/data/esdt"
+)
+
+// ErrNotESDTTransferInput signals invalid ESDT transfer input error
+var ErrNotESDTTransferInput = errors.New("not ESDT transfer input")
+
+// ErrNotEnoughArguments signals not enough arguments error
+var ErrNotEnoughArguments = errors.New("not enough arguments")
+
+// ErrNilMarshalizer signals that marshalizer is nil
+var ErrNilMarshalizer = errors.New("nil marshalizer")
+
+// MinArgsForESDTTransfer defines the minimum arguments needed for an esdt transfer
+const MinArgsForESDTTransfer = 2
+
+// MinArgsForESDTNFTTransfer defines the minimum arguments needed for an nft transfer
+const MinArgsForESDTNFTTransfer = 4
+
+// MinArgsForMultiESDTNFTTransfer defines the minimum arguments needed for a multi transfer
+const MinArgsForMultiESDTNFTTransfer = 4
+
+// ArgsPerTransfer defines the number of arguments per transfer in multi transfer
+const ArgsPerTransfer = 3
+
+type esdtTransferParser struct {
+	marshalizer vmcommon.Marshalizer
+}
+
+// NewESDTTransferParser creates a new esdt transfer parser
+func NewESDTTransferParser(
+	marshalizer vmcommon.Marshalizer,
+) (*esdtTransferParser, error) {
+	if check.IfNil(marshalizer) {
+		return nil, ErrNilMarshalizer
+	}
+
+	return &esdtTransferParser{marshalizer: marshalizer}, nil
+}
+
+// ParseESDTTransfers returns the list of esdt transfers, the callFunction and callArgs from the given arguments
+func (e *esdtTransferParser) ParseESDTTransfers(
+	sndAddr []byte,
+	rcvAddr []byte,
+	function string,
+	args [][]byte,
+) (*vmcommon.ParsedESDTTransfers, error) {
+	switch function {
```

Comment:
we can avoid this switch by further splitting this implementation in smaller, dedicated sub-parsers.

Reply:
yeah and a container for that and a get from container. I would leave like this for now.